src/API/OnboardingTasks.php

```diff
@@ -736,4 +788,51 @@ public function undo_dismiss_task( $request ) {
 
 		return rest_ensure_response( $update );
 	}
+
+	/**
+	 * Snooze an onboarding task.
+	 *
+	 * @param WP_REST_Request $request Request data.
+	 *
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function snooze_task( $request ) {
+
+		$task_id         = $request->get_param( 'task_id' );
+		$task_list_id    = $request->get_param( 'task_list_id' );
+		$snooze_duration = $request->get_param( 'duration' );
+
+		$is_snoozeable = false;
+
+		$snooze_task = OnboardingTasksFeature::get_task_by_id( $task_id, $task_list_id );
+
+		if ( $snooze_task && isset( $snooze_task['isSnoozeable'] ) && $snooze_task['isSnoozeable'] ) {
+			$is_snoozeable = true;
+		}
+
+		if ( ! $is_snoozeable ) {
+			return new \WP_Error(
+				'woocommerce_tasks_invalid_task',
+				__( 'Sorry, no snoozeable task with that ID was found.', 'woocommerce-admin' )
+			);
+		}
+
+		$snooze_option = get_option( 'woocommerce_task_list_remind_me_later_tasks' );
```

Comment:
Needs default param to avoid error on initial snooze:

```suggestion
		$snooze_option = get_option( 'woocommerce_task_list_remind_me_later_tasks', array() );
```

Reply:
Good catch 👍🏻